code/src/cljs/sixsq/nuvla/ui/credentials/views.cljs

```diff
@@ -790,6 +905,19 @@
            [ui/Image {:src   "/ui/images/gpg.png"
                       :style {:max-height 112}}]]]
 
+         [ui/Card
+          {:on-click #(do
```

Comment:
Should it be one dispatch?

Reply:
Yes it should be in the best of the worlds. Changing this to single dispatch that do other dispatches will not enhance much since there is easier ways to validate forms. This page needs a refactor but it's not our priority for now. We can make an epic to discuss how we can enhance the code quality of this page, adding pagination, form validation, single dispatch, make the add credential modal smaller because now it's too big with too much cards, perhaps categorizing it, etc.